writeengine/server/we_dmlcommandproc.cpp

```diff
@@ -4468,6 +4472,7 @@ uint8_t WE_DMLCommandProc::processFixRows(messageqcpp::ByteStream& bs,
                 colStruct.colWidth = 8;
                 colStruct.tokenFlag = true;
                 dctnryStruct.dctnryOid = colType.ddn.dictOID;
+                dctnryStruct.fCharsetNumber = colType.charsetNumber;
```

Comment:
It makes sense to assign fCharsetNumber in the non-token case also.
I would suggest to add and use a new WriteEngine::DctnryStruct that takes all the params as its arguments.